sound/soc/sof/core.c

```diff
@@ -242,6 +245,25 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 	if (!sof_ops(sdev)->runtime_suspend || !sof_ops(sdev)->runtime_resume)
 		pm_runtime_get_noresume(sdev->dev);
 
+	/* Now register virtbus devices */
+	sdev->sof_virtbus_dev = devm_kzalloc(sdev->dev,
+					     sizeof(*sdev->sof_virtbus_dev),
+					     GFP_KERNEL);
+	if (!sdev->sof_virtbus_dev)
+		return -ENOMEM;
+
+	sdev->sof_virtbus_dev->sdev = sdev;
```

Comment:
do you really need this circular pointer loop?

why not add a struct virtbus_device in sdev and use container_of to get to sdev?

and then you'd do a virtbus_dev_register(&sdev->virtbus_dev)

Reply:
@plbossart yes i could just add virtbus_device but then we are going to need to add more if these devices and it will be an array. I could do one of 2 things: just have an array of virtbus_device or extend the virtbus_device definition to add data to it but not sure we really need to.

Reply:
I've not read the virtio patches yet (sorry, I know I should), but wouldn't sdev->dev(physical dev) be the parent for each virtbus_device (rather than a container) ?   Being a parent device would keep the runtime PM refcounting working too.